Added local telemetry file and survey details. (#8654)
flutter/website: src/reference/crash-reporting.md

Google handles all data reported by this tool in accordance with the 
[Google Privacy Policy][].

> Additionally, you may review the recently reported data in the 
> `.dart-tool/dart-flutter-telemetry.log` file under the home directory
> `~/` (macOS/Linux) or the `<username>\AppData` (Windows) directory. 

Reviewer: **issue:** This points to the wrong Windows directory. The macOS directory is correct. Make this change and I can approve.
```suggestion
You may review the recently reported data in the 
`.dart-tool/dart-flutter-telemetry.log` file.
In macOS or Linux, this log is located in the home directory (`~/`).
In Windows, this log is located in the Roaming AppData directory in Windows (`%APPDATA%`). 
```

Reviewer 2: Nice catch to you use the env variable, that's actually how we do it in `package:unified_analytics` as well

```dart
Directory getHomeDirectory(FileSystem fs) {
  String? home;
  Map<String, String> envVars = io.Platform.environment;

  if (io.Platform.isMacOS) {
    home = envVars['HOME'];
  } else if (io.Platform.isLinux) {
    home = envVars['HOME'];
  } else if (io.Platform.isWindows) {
    home = envVars['AppData'];
  }

  return fs.directory(home!);
}
```

Google handles all data reported by this tool in accordance with the 
[Google Privacy Policy][].

Additionally, you may review the recently reported data in the 
`.dart-tool/dart-flutter-telemetry.log` file under the home directory

> `~/` (macOS/Linux) or the `<username>\AppData` (Windows) directory. 

Reviewer: **question:** When does this directory get created? I checked my macOS and Windows systems and neither had the file you mention, nor the directory at that location. `.dart-tool` is in each Flutter app directory, but not at the top level directory.

Reviewer 2: This directory along with the files should be getting created in through the latest stable release for Flutter and Dart from i/o. Specifically, we have a new analytics package, `package:unified_analytics` that is responsible for creating these new files.

Once you run anything with the latest flutter cli (ie. `flutter`) or open up VS Code into a dart project (need to have the dart extensions installed), you should see that directory getting created

## Disabling analytics reporting

> You can opt out of anonymous crash reporting and feature
> usage statistics from Flutter by running the following command:

Reviewer: **issue:** Condition, then instruction per [GSG](https://developers.google.com/style/sentence-structure).
```suggestion
To opt out of anonymous crash reporting and feature
usage statistics, run the following command:
```

```terminal
$ flutter --disable-telemetry
```

> If you opt out of analytics, an opt-out event will be sent,
> and then no further information will be stored or sent by that
> installation of Flutter. 

Reviewer: **Issue:** Split sentences to ease readability.
```suggestion
If you opt out of analytics, Flutter sends an opt-out event.
This Flutter installation neither sends nor stores any further information. 
```

Author: Committed the change!

Google handles all data reported by this tool in accordance with the 
[Google Privacy Policy][].

Additionally, you may review the recently reported data in the 
`dart-flutter-telemetry.log` file on your disk.

> ## Disabling analytics reporting

Reviewer 2: I would mention it is also located in the user's HOME directory under the `.dart-tool` directory.. for example on macOS, you can find it at `$HOME/.dart-tool/dart-flutter-telemetry.log`

Author: Added!

> ```

Reviewer 2: Additionally, we should change the command they need to run to
```
flutter --disable-telemetry
``` 
The command above will work for both the old and new unified analytics package but the PDD states we should use the one above

And to re-enable incase they want to is below
```
flutter --enable-telemetry
```

Author: Thanks, updated! Will `flutter config` still work?

Reviewer 2: Perfect, and yep that will still work so we don't break any previous users but we should encourage the use of the new command for telemetry related stuff
